$MirOS: ports/shells/zsh/patches/patch-Src_Zle_zle_refresh_c,v 1.1 2009/09/06 16:31:12 tg Exp $
--- Src/Zle/zle_refresh.c.orig	Mon Dec 20 09:51:20 2010
+++ Src/Zle/zle_refresh.c	Tue Mar  8 17:18:23 2011
@@ -2508,9 +2508,12 @@ singlerefresh(ZLE_STRING_T tmpline, int 
 	    wchar_t wc;
 	    int started = 0;
 
+#if !defined(__WCHAR_MAX__) || (__WCHAR_MAX__ > 65535U)
 	    if ((unsigned)tmpline[t0] > 0xffffU) {
 		sprintf(dispchars, "<%.08x>", (unsigned)tmpline[t0]);
-	    } else {
+	    } else
+#endif
+	      {
 		sprintf(dispchars, "<%.04x>", (unsigned)tmpline[t0]);
 	    }
 	    while (*dispptr) {
